src/status_im2/contexts/syncing/enter_sync_code/view.cljs

```diff
@@ -0,0 +1,76 @@
+(ns status-im2.contexts.syncing.enter-sync-code.view
+  (:require [clojure.string :as string]
+            [quo2.core :as quo]
+            [quo2.foundations.colors :as colors]
+            [reagent.core :as reagent]
+            [react-native.core :as rn]
+            [react-native.clipboard :as clipboard]
+            [utils.i18n :as i18n]
+            [status-im2.contexts.syncing.enter-sync-code.style :as style]
+            [utils.debounce :as debounce]
+            [utils.re-frame :as rf]
+            [status-im2.contexts.syncing.utils :as sync-utils]))
+
+(defn view
+  []
+  (let [default-value           (reagent/atom "")
+        submit-button-disabled? (reagent/atom (string/blank? default-value))
+        profile-color           (rf/sub [:profile/customization-color])]
+    (fn []
+      (let [invalid?           false
+            show-paste-button? (string/blank? @default-value)]
```

Comment:
I believe we should also check whether the clipboard has any value in it.

Reply:
we use `show-paste-button?` let binding to determine whether to show past button or the clear icon.
We show the clear icon when there is any text in it otherwise we default to past button.

Now it would be nice to only show paste button if there is some value in clipboard however that would come at a cost of checking clipboard at component render time which in turn will show the IOS native modal.
ref : screenshot below

![Screenshot 2023-08-07 at 3 27 16 PM](https://github.com/status-im/status-mobile/assets/64726664/cf18684f-2e35-4e89-9a30-67e4ee91acda)

I think we are better off without this and once someone taps the paste button with an empty clipboard the app would do nothing, what do you think ?

Reply:
we can also go back to design team and discuss about this case and implement that in a follow up since its a UX related decision, for now I think this is okay :D

Reply:
I have no strong opinion on this. As a user, Showing the `paste` button with empty clipboard is misleading. As a developer, we certainly can't avoid the permission dialog, and meantime we can't show the `paste` button with an empty clipboard.  The `Add new contact` screen/sheet does this check, so I referred to that behaviour here.

Reply:
Hi @smohamedjavid : I just discussed this with Design and they want an always enabled `paste` button.
Will create an issue to update the behaviour in contacts screen as well

Reply:
Thanks @siddarthkay!